vmas/simulator/dynamics/drone.py

```diff
@@ -0,0 +1,142 @@
+from typing import Union
+
+import torch
+from torch import Tensor
+import vmas.simulator.core
+import vmas.simulator.utils
+from vmas.simulator.dynamics.common import Dynamics
+
+
+class Drone(Dynamics):
+    def __init__(
+        self,
+        world: vmas.simulator.core.World,
+        I_xx: float = 8.1e-3,
+        I_yy: float = 8.1e-3,
+        I_zz: float = 14.2e-3,
+        integration: str = "rk4",
+    ):
+        super().__init__()
+
+        assert integration in (
+            "rk4",
+            "euler",
+        )
+        
+        self.integration = integration
+        self.I_xx = I_xx
+        self.I_yy = I_yy
+        self.I_zz = I_zz
+        self.world = world
+        self.g = 9.81
+        self.dt = world.dt
+        self.reset()
+        
+    def reset(self, index: Union[Tensor, int] = None):
+        if index is None:
+            # Drone state: phi(roll), theta (pitch), psi (yaw),
+            #              p (roll_rate), q (pitch_rate), r (yaw_rate),
+            #              x_dot (vel_x), y_dot (vel_y), z_dot (vel_z),
+            #              x (pos_x), y (pos_y), z (pos_z)
+            self.drone_state = torch.tensor(
+                [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0], device=self.world.device
+            ).unsqueeze(0).repeat(self.world.batch_dim, 1)
+        else:
+            self.drone_state[index] = 0.0
+            
+    def need_reset(self) -> bool:
+        # Constraint roll and pitch within +-30 degrees
+        if torch.any(torch.abs(self.drone_state[:, 0:2]) > 30 * (torch.pi / 180)):
+            return True
+
+        return False
```

Comment:
```suggestion
    def need_reset(self) -> bool:
        # Constraint roll and pitch within +-30 degrees
        return torch.any(self.drone_state[:, :2].abs() > 30 * (torch.pi / 180), dim=-1)
 
```

You want to reset if the abs value of any of the first 2 dims is grater than that right?

Reply:
Yeah. The idea to constraint the roll or pitch angle of the drone between -30 and 30 degrees, so the drone will not flip over.